// @irembo/irembopay-node-sdk/dist/base.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Base = void 0;
const axios_1 = __importDefault(require("axios"));
class Base {
    constructor(apiKey, environment) {
        this.apiKey = apiKey;
        switch (environment) {
            case "sandbox":
                this.baseURL = "https://api.sandbox.irembopay.com/payments";
                break;
            case "checkout":
                this.baseURL = "https://api.checkout.irembopay.com/payments";
                break;
            default:
                this.baseURL = "https://api.irembopay.com/payments";
                break;
        }
        this.axiosInstance = axios_1.default.create({
            baseURL: this.baseURL,
            headers: {
                "irembopay-secretkey": this.apiKey,
                "X-API-Version": 2,
                "User-Agent": "iremboPay-SDK",
                Accept: "application/json",
                "Content-Type": "application/json",
            },
        });
    }
    handleError(error) {
        // Check if error.response is defined before accessing it
        if (error.response) {
            throw error.response.data || error.response;
        }
        else {
            throw error;
        }
    }
}
exports.Base = Base;
