// @irembo/irembopay-node-sdk/src/base.ts
import axios, { AxiosInstance } from "axios";

export abstract class Base {
  private apiKey: string;
  private baseURL: string;
  protected axiosInstance: AxiosInstance;

  constructor(apiKey: string, environment: Environment) {
    this.apiKey = apiKey;
    switch (environment) {
      case "sandbox":
        this.baseURL = "https://api.sandbox.irembopay.com/payments";
        break;

      case "checkout":
        this.baseURL = "https://api.checkout.irembopay.com/payments";
        break;

      default:
        this.baseURL = "https://api.irembopay.com/payments";
        break;
    }
    this.axiosInstance = axios.create({
      baseURL: this.baseURL,
      headers: {
        "irembopay-secretkey": this.apiKey,
        "X-API-Version": 2,
        "User-Agent": "iremboPay-SDK",
        Accept: "application/json",
        "Content-Type": "application/json",
      },
    });
  }
  protected handleError(error: any): void {
    // Check if error.response is defined before accessing it
    if (error.response) {
      throw error.response.data || error.response;
    } else {
      throw error;
    }
  }
}
